geniza/corpus/management/commands/tei_to_annotation.py

```diff
@@ -0,0 +1,347 @@
+    """Synchronize TEI transcriptions to edition footnote content"""
+
+    v_normal = 1  # default verbosity
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "files", nargs="*", help="Only convert the specified files."
+        )
+
+    def handle(self, *args, **options):
+
+        # generate or retrieve a token for script user to use for token-auth
+        token = Token.objects.get_or_create(
+            user=User.objects.get(username=settings.SCRIPT_USERNAME)
+        )[0]
+
+        # get settings for remote git repository url and local path
+        gitrepo_url = settings.TEI_TRANSCRIPTIONS_GITREPO
+        gitrepo_path = settings.TEI_TRANSCRIPTIONS_LOCAL_PATH
+
+        self.verbosity = options["verbosity"]
+        # get content type and script nuser for log entries
+        self.footnote_contenttype = ContentType.objects.get_for_model(Footnote)
+        self.script_user = User.objects.get(username=settings.SCRIPT_USERNAME)
+
+        # NOTE: some overlap with sync transcriptions manage command
+
+        # make sure we have latest tei content from git repository
+        # (inherited from sync transcriptions command)
+        self.sync_git(gitrepo_url, gitrepo_path)
+
+        self.stats = defaultdict(int)
+
+        xmlfiles = options["files"] or glob.iglob(os.path.join(gitrepo_path, "*.xml"))
+
+        annotation_client = AnnotationStore(auth_token=token.key)
+
+        # iterate through tei files to be migrated
+        for xmlfile in xmlfiles:
+            self.stats["xml"] += 1
+            if self.verbosity >= self.v_normal:
+                self.stdout.write(xmlfile)
+
+            xmlfile_basename = os.path.basename(xmlfile)
+
+            tei = xmlmap.load_xmlobject_from_file(xmlfile, GenizaTei)
+            # some files are stubs with no content
+            # check if the tei is ok to proceed; (e.g., empty or only translation content)
+            # if empty, report and skip
+            if not self.check_tei(tei, xmlfile):
+                self.stdout.write(
+                    self.style.WARNING(
+                        "No transcription content in %s; skipping" % xmlfile
+                    )
+                )
+                continue
+            # get the document for the file based on id / old id
+            doc = self.get_pgp_document(xmlfile_basename)
+            # if document was not found, skip
+            if not doc:
+                self.stdout.write(
+                    self.style.WARNING("Document not found for %s; skipping" % xmlfile)
+                )
+                continue
+            # found the document
+            if self.verbosity >= self.v_normal:
+                self.stdout.write(doc)
+
+            # get the footnote for this file & doc
+            footnote = self.get_edition_footnote(doc, tei, xmlfile)
+            # if no footnote, skip for now
+            # TODO: generate footnote when needed! (but should we need to?)
```

Comment:
Is there a real-world condition that would cause this to happen?

Reply:
I'm running into it in my testing, but I don't think my data is up to date; was going to refresh and see if it still happens.